crates/network-scanner/src/mdns.rs

```diff
@@ -0,0 +1,86 @@
+use mdns_sd::ServiceEvent;
+
+use crate::{
+    task_utils::{PortReceiver, TaskManager},
+    ScannerError,
+};
+
+const META_QUERY: &str = "_services._dns-sd._udp.local.";
+
+pub fn mdns_query_scan(
+    service_deamon: mdns_sd::ServiceDaemon,
+    task_manager: TaskManager,
+    entire_duration: std::time::Duration,
+    single_query_duration: std::time::Duration,
+) -> Result<PortReceiver, ScannerError> {
+    let receiver = service_deamon.browse(META_QUERY)?;
+    let service_deamon_clone = service_deamon.clone();
+
+    let (result_sender, result_receiver) = tokio::sync::mpsc::channel(255);
+
+    task_manager
+        .with_timeout(entire_duration)
+        .when_finished(move || {
+            tracing::debug!("mdns meta query finished");
+            while let Err(e) = service_deamon_clone.stop_browse(META_QUERY) {
+                if let mdns_sd::Error::Again = e {
+                    continue;
+                }
+                tracing::error!("mdns stop browse error: {}", e);
+            }
```

Comment:
issue: The loop logic seems erroneous. My understanding is that we should _break_ when the variant is not `Again`, and otherwise the `continue` keyword is not even required since this is a `while` loop.

nitpick: Use tracing structured logging, especially emit an `error` field for errors.

```suggestion
            while let Err(e) = service_deamon_clone.stop_browse(META_QUERY) {
                match e {
                    mdns_sd::Error::Again => {
                        tracing::trace!("mdns stop_browse transient error, trying again");
                    }
                    fatal => {
                        tracing::error!(error = %fatal, "mdns stop_browse fatal error");
                        break;
                    }
                }
            }
```